lib/modules/datasource/rubygems/index.ts

```diff
@@ -91,102 +89,6 @@ export class RubyGemsDatasource extends Datasource {
     }
   }
 
-  @cache({
-    namespace: `datasource-${RubyGemsDatasource.id}`,
-    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
-      // TODO: types (#7154)
-      /* eslint-disable @typescript-eslint/restrict-template-expressions */
-      `metadata:${registryUrl}/${packageName}`,
-  })
-  async fetchGemMetadata(
-    registryUrl: string,
-    packageName: string
-  ): Promise<GemMetadata | null> {
-    try {
-      const { body } = await this.http.getJson(
-        joinUrlParts(registryUrl, '/api/v1/gems', `${packageName}.json`),
-        GemMetadata
-      );
-      return body;
-    } catch (err) {
-      // fallback to deps api on 404
-      if (err instanceof HttpError && err.response?.statusCode === 404) {
-        return null;
-      }
-      throw err;
-    }
-  }
-
-  @cache({
-    namespace: `datasource-${RubyGemsDatasource.id}`,
-    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
-      // TODO: types (#7154)
-      /* eslint-disable @typescript-eslint/restrict-template-expressions */
-      `versions:${registryUrl}/${packageName}`,
-  })
-  async fetchGemVersions(
-    registryUrl: string,
-    packageName: string
-  ): Promise<GemVersions | null> {
-    try {
-      const { body } = await this.http.getJson(
-        joinUrlParts(registryUrl, '/api/v1/versions', `${packageName}.json`),
-        GemVersions
-      );
-      return body;
-    } catch (err) {
-      if (err.statusCode === 400 || err.statusCode === 404) {
-        logger.debug(
-          { registry: registryUrl },
-          'versions endpoint returns error - falling back to info endpoint'
-        );
-        return null;
-      } else {
-        throw err;
-      }
-    }
-  }
-
-  async getReleasesViaAPI(
-    registryUrl: string,
-    packageName: string,
-    gemMetadata: GemMetadata
-  ): Promise<ReleaseResult | null> {
-    const gemVersions = await this.fetchGemVersions(registryUrl, packageName);
-
-    let releases: Release[] | null = null;
-    if (gemVersions?.length) {
-      releases = gemVersions;
-    } else if (gemMetadata.latestVersion) {
-      releases = [{ version: gemMetadata.latestVersion }];
-    } else {
-      return null;
-    }
-
-    const result: ReleaseResult = { releases };
-
-    if (gemMetadata.changelogUrl) {
-      result.changelogUrl = gemMetadata.changelogUrl;
-    }
-
-    if (gemMetadata.homepage) {
-      result.homepage = gemMetadata.homepage;
-    }
-
-    if (gemMetadata.sourceUrl) {
-      result.sourceUrl = gemMetadata.sourceUrl;
-    }
-
-    return result;
-  }
-
-  @cache({
```

Comment:
why cache removal?

Reply:
Rubygems data is cached in separate place, and we don't want cache other registries, e.g. private ones